tests/zfs-tests/tests/functional/xattr/xattr_011_pos.ksh

```diff
@@ -172,66 +173,75 @@ else
 	log_must pax -r -f $TESTDIR/xattr.pax $TESTDIR
 	log_mustnot eval "runat $TESTDIR/pax.$$ cat passwd > /dev/null 2>&1"
 	log_must rm $TESTDIR/pax.$$ $TESTDIR/noxattr.pax $TESTDIR/xattr.pax
+else
+	log_note "Checking pax - unsupported"
 fi
 
 log_note "Checking tar"
-if is_linux; then
+if is_illumos; then
 	log_must touch $TESTDIR/tar.$$
 	create_xattr $TESTDIR/tar.$$ passwd /etc/passwd
 
 	log_must cd $TESTDIR
 
-	log_must tar -cf noxattr.tar tar.$$
-	log_must tar --xattrs -cf xattr.tar tar.$$
+	log_must tar cf noxattr.tar tar.$$
+	log_must tar c@f xattr.tar tar.$$
 	log_must rm $TESTDIR/tar.$$
 
 	# we should have no xattr here
-	log_must tar --no-xattrs -xf xattr.tar
-	log_mustnot get_xattr passwd $TESTDIR/tar.$$
+	log_must tar xf xattr.tar
+	log_mustnot eval "runat $TESTDIR/tar.$$ cat passwd > /dev/null 2>&1"
 	log_must rm $TESTDIR/tar.$$
 
 	# we should have an xattr here
-	log_must tar --xattrs -xf xattr.tar
+	log_must tar x@f xattr.tar
 	verify_xattr tar.$$ passwd /etc/passwd
 	log_must rm $TESTDIR/tar.$$
 
 	# we should have no xattr here
-	log_must tar --no-xattrs -xf $TESTDIR/noxattr.tar
-	log_mustnot get_xattr passwd $TESTDIR/tar.$$
+	log_must tar xf $TESTDIR/noxattr.tar
+	log_mustnot eval "runat $TESTDIR/tar.$$ cat passwd > /dev/null 2>&1"
 	log_must rm $TESTDIR/tar.$$
 
 	# we should have no xattr here
-	log_must tar --xattrs -xf $TESTDIR/noxattr.tar
-	log_mustnot get_xattr passwd $TESTDIR/tar.$$
+	log_must tar x@f $TESTDIR/noxattr.tar
+	log_mustnot eval "runat $TESTDIR/tar.$$ cat passwd > /dev/null 2>&1"
 	log_must rm $TESTDIR/tar.$$ $TESTDIR/noxattr.tar $TESTDIR/xattr.tar
 else
 	log_must touch $TESTDIR/tar.$$
 	create_xattr $TESTDIR/tar.$$ passwd /etc/passwd
 
 	log_must cd $TESTDIR
 
-	log_must tar cf noxattr.tar tar.$$
-	log_must tar c@f xattr.tar tar.$$
+	if is_freebsd; then
+		# FreeBSD tar has xattrs by default.
+		log_must tar --no-xattrs -cf noxattr.tar tar.$$
+		log_must tar -cf xattr.tar tar.$$
+	else
+		# GNU tar does not have xattrs by default.
+		log_must tar -cf noxattr.tar tar.$$
+		log_must tar --xattrs -cf xattr.tar tar.$$
+	fi
```

Comment:
These can be combined by being explicit in both cases, I'll fix that when I update zts-report.py.